nucypher/characters/lawful.py

```diff
@@ -925,21 +892,11 @@ def _get_cleartexts(self,
 
     def _handle_treasure_map(self,
                              publisher_verifying_key: PublicKey,
-                             label: bytes,
-                             encrypted_treasure_map: Optional['EncryptedTreasureMap'] = None,
-                             ) -> 'TreasureMap':
-        if encrypted_treasure_map:
-            treasure_map = self._decrypt_treasure_map(encrypted_treasure_map, publisher_verifying_key)
-            self.treasure_maps[treasure_map.hrac] = treasure_map
-        else:
-            hrac = self.construct_policy_hrac(publisher_verifying_key, label)
-            try:
-                treasure_map = self.treasure_maps[hrac]
-            except KeyError:
-                # If the treasure map is not known, join the policy as part of retrieval.
-                self.join_policy(label=label, publisher_verifying_key=publisher_verifying_key)
-                treasure_map = self.treasure_maps[hrac]
-
+                             encrypted_treasure_map: Optional['EncryptedTreasureMap']
+                             ) -> Tuple['TreasureMap', int]:
+        """Decrypt and cache the treasure map by Bob."""
+        treasure_map: TreasureMap = self._decrypt_treasure_map(encrypted_treasure_map, publisher_verifying_key)
```

Comment:
Should we check if it's in cache first? 

On a second thought, now that we're getting a treasure map anyway, how much are we saving by having a cache? A decryption time?

Reply:
✔️